src/crypto/DeviceList.js

```diff
@@ -364,13 +383,24 @@ export default class DeviceList {
     /**
      * Find a device by curve25519 identity key
      *
-     * @param {string} userId     owner of the device
+     * @param {string} userId     owner of the device (optional)
```

Comment:
i'm completely blanking on why we bother letting you specify userId at all, if it's optional - given we can work it out from senderKey.  i assume there's an excellent reason,  but it should probably be spelt out.  it also feels really weird to  put the optional param at the beginning of the arg list and do this shift dance rather than having it as an optional last arg.

Reply:
Turns out there was only one other place it was called anyway so I have just made it the 2 arg version